sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRegExpReplaceMeta.scala

```diff
@@ -51,6 +53,8 @@ class GpuRegExpReplaceMeta(
     expr.regexp match {
       case Literal(s: UTF8String, DataTypes.StringType) if s != null =>
         javaPattern = Some(s.toString())
+        // check that this is valid in Java
+        Pattern.compile(javaPattern.toString)
```

Comment:
I think we should move this to the RegexTranspiler code. I think it could be done in both `getTranspiledAST(...)`/`RegexParser(...)` 

Otherwise this line will have to be called in every Regexp Expression class, and it could easily be lost in a few places. The transpiler is used by all of these methods, so this makes sense as a shortcut.

Reply:
The validity check is now incorporated into the `Parser(string).parse()` function. This ensures the check is performed once when the string is converted to `RegexAST`. Consequently, it is also invoked for `getTranspiledAST`.

```
def parse(): RegexAST = {
    // Validate if the pattern is compatible with Java, as this would throw an error otherwise
    Pattern.compile(pattern)
...
}
```